src/biogeochem/CNPhenologyMod.F90

```diff
@@ -2309,26 +2495,70 @@ subroutine PlantCrop(p, leafcn_in, jday, &
          harvdate          =>    crop_inst%harvdate_patch                        , & ! Output: [integer  (:) ]  harvest date
+         next_rx_sdate     =>    crop_inst%next_rx_sdate                         , & ! Inout:  [integer  (:) ]  prescribed sowing date of next growing season this year
          sowing_count      =>    crop_inst%sowing_count                          , & ! Inout:  [integer  (:) ]  number of sowing events this year for this patch
-         idop              =>    cnveg_state_inst%idop_patch                     , & ! Output: [integer  (:) ]  date of planting                                   
+         sowing_reason     =>    crop_inst%sowing_reason_thisyr                  , & ! Output:  [real(r8)  (:) ]  reason for each sowing this year for this patch
+         gddmaturity       =>    cnveg_state_inst%gddmaturity_patch            , & ! Output: [real(r8) (:) ]  gdd needed to harvest
+         idop              =>    cnveg_state_inst%idop_patch                     , & ! Output: [integer  (:) ]  date of planting
+         iyop              =>    cnveg_state_inst%iyop_patch                     , & ! Output: [integer  (:) ]  year of planting
          leafc_xfer        =>    cnveg_carbonstate_inst%leafc_xfer_patch         , & ! Output: [real(r8) (:) ]  (gC/m2)   leaf C transfer
          leafn_xfer        =>    cnveg_nitrogenstate_inst%leafn_xfer_patch       , & ! Output: [real(r8) (:) ]  (gN/m2)   leaf N transfer
          crop_seedc_to_leaf =>   cnveg_carbonflux_inst%crop_seedc_to_leaf_patch  , & ! Output: [real(r8) (:) ]  (gC/m2/s) seed source to leaf
-         crop_seedn_to_leaf =>   cnveg_nitrogenflux_inst%crop_seedn_to_leaf_patch & ! Output: [real(r8) (:) ]  (gN/m2/s) seed source to leaf
+         crop_seedn_to_leaf =>   cnveg_nitrogenflux_inst%crop_seedn_to_leaf_patch, & ! Output: [real(r8) (:) ]  (gN/m2/s) seed source to leaf
+         hybgdd            =>    pftcon%hybgdd                                 , & ! Input:  [real(r8) (:) ]
+         gddmin            =>    pftcon%gddmin                                 , & ! Input:
+         gdd020            =>    temperature_inst%gdd020_patch                 , & ! Input:  [real(r8) (:) ]  20 yr mean of gdd0
+         gdd820            =>    temperature_inst%gdd820_patch                 , & ! Input:  [real(r8) (:) ]  20 yr mean of gdd8
+         gdd1020           =>    temperature_inst%gdd1020_patch                  & ! Input:  [real(r8) (:) ]  20 yr mean of gdd10
          )
 
       ! impose limit on growing season length needed
       ! for crop maturity - for cold weather constraints
       croplive(p)  = .true.
       idop(p)      = jday
+      iyop(p)      = kyr
       harvdate(p)  = NOT_Harvested
-      sowing_count(p) = sowing_count(p) + 1
-      crop_inst%sdates_thisyr(p,sowing_count(p)) = jday
+      s = sowing_count(p) + 1
+
+      sowing_count(p) = s
+      if (s < mxsowings) then
+         next_rx_sdate(p) = crop_inst%rx_sdates_thisyr(p, s+1)
```

Comment:
- [ ] It seems like there is something implicit that sets next_rx_sdate to something like -1 if not using prescribed sowing dates. It would help to make this explicit either in a comment here or by adding an explicit conditional on whether we're using prescribed sowing dates.

Reply:
Resolved by cd904fc0a50e.